rescheduler/rescheduler.go

```diff
@@ -17,18 +17,28 @@ limitations under the License.
 package main
 
 import (
+	"encoding/json"
+	"fmt"
 	"os"
 	"time"
 
+	ca_simulator "k8s.io/contrib/cluster-autoscaler/simulator"
 	kube_utils "k8s.io/contrib/cluster-autoscaler/utils/kubernetes"
 	kube_api "k8s.io/kubernetes/pkg/api"
 	kube_client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
 	kubectl_util "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
 
 	"github.com/golang/glog"
 	flag "github.com/spf13/pflag"
 )
 
+const (
+	criticalAnnotation = "scheduler.alpha.kubernetes.io/critical-pod"
+	criticalAddonsOnly = "CriticalAddonsOnly"
```

Comment:
Unrelated (but Github won't let me comment on a line you didn't change...):

`inCluster = flags.Bool("running-in-cluster", true, ...`

Where would the controller be running if not in a cluster? (One someone's laptop?) When do you set this to false?